### ISSUE TITLE
CardinalComm SDK Warning from Google Play

### ISSUE THREAD

**Issue by dimitradg** [braintree/braintree_android#629]
**Integration Details:**

* SDK/Library version: 4.12.0
* Environment: Production
* Android Version and Device: All
* Braintree dependencies:
gradle (app level)

implementation 'com.braintreepayments.api:card:4.12.0'
implementation 'com.braintreepayments.api:data-collector:4.12.0'
implementation 'com.braintreepayments.api:three-d-secure:4.12.0'

app (project level)

`maven  {
            url "https://cardinalcommerceprod.jfrog.io/artifactory/android"
            credentials {
                username ****
                password ****
            }
        }`

**Describe the bug**
I am using the above mentioned brain tree sdks (v4.12.0) and today I received a warning from Google Play with the following message:
Your app is using the Cardinal Mobile SDK which collects persistent device identifiers, but may not be limited to Android ID and hardware IDs. Persistent device identifiers may not be linked to any other personal and sensitive user data and/or resettable device identifiers (e.g., IMEI, IMSI, SIM Serial #, etc).

You may consider upgrading to a policy-compliant version of this SDK, if available from your SDK provider or removing the SDK. According to your SDK provider, you may consider upgrading to 2.2.7-2. Please consult the SDK provider for further information. Google is unable to endorse or recommend any third party software.

Thank you in advance

**Comment by SirLordPouya** [comment]
Hello, we also got this email today. I'm not sure if it helps, but after checking the build scan, the resolved Cardinal dependency version is 2.2.7-2 for our project, but we still got that email. Maybe the latest version of cardinal (2.2.7-2) also is not policy-compliant?

**Comment by dimitradg** [comment]
> Hello, we also got this email today. I'm not sure if it helps, but after checking the build scan, the resolved Cardinal dependency version is 2.2.7-2 for our project, but we still got that email. Maybe the latest version of cardinal (2.2.7-2) also is not policy-compliant?


@SirLordPouya May I ask where did you manage to check cardinal dependency? This may be a silly question but I am kind of confused.

**Comment by SirLordPouya** [comment]
> @SirLordPouya May I ask where did you manage to check cardinal dependency? This may be a silly question but I am kind of confused.

You can use [gradle build scan](https://scans.gradle.com/?_ga=2.53889192.1888159940.1669885169-628957025.1667850982). `gradle build --scan`

**Comment by marxhendrik** [comment]
Exact same issue here. We are using this version:

> According to your SDK provider, you may consider upgrading to 2.2.7-2

**Comment by dimitradg** [comment]
> > @SirLordPouya May I ask where did you manage to check cardinal dependency? This may be a silly question but I am kind of confused.
> 
> @dimitradg You can use [gradle build scan](https://scans.gradle.com/?_ga=2.53889192.1888159940.1669885169-628957025.1667850982). `gradle build --scan`

Thank you :)

**Comment by dimitradg** [comment]
> Exact same issue here. We are using this version:
> 
> > According to your SDK provider, you may consider upgrading to 2.2.7-2

I saw the issue you commended above - however it is not very clear to me I have to change the content of Data Safety form and what should I change

**Comment by bm185178** [comment]
What's the concrete step for this? Would it be resolving the warning if we upgrade braintree sdk which supports 2.2.7-2? This will suspend our customers' releases in the future. This should not be taken lightly because dealing with google suspension is not easy.

**Comment by 4gus71n** [comment]
+1 ☝️ – I'm in the same spot, we just got this warning from Google and we have 60 days to fix this issue otherwise our app will be taken off the Google Playstore. Please share some updates as soon as there's a fix for this. Thanks! 🙇

**Comment by sshropshire** [comment]
Thanks all. We're investigating the issue and should have an update in the coming days. We appreciate your patience in the meantime.

**Comment by hollabaq86** [comment]
Hi folks, can I get 👍 reactions from folks getting these notifications that:
1. You are on the latest version of the SDK (4.20.0)
2. The text of the notification from the Google Playstore specifically mentions the following:
> Your app is using the Cardinal Mobile SDK which collects persistent device identifiers, but may not be limited to Android ID and hardware IDs. Persistent device identifiers may not be linked to any other personal and sensitive user data and/or resettable device identifiers (eg, IMEI, IMSI, SIM Serial #, etc).

If the contents of the notification you're getting from Google is different, please drop that in a comment here or [contact Support](https://developer.paypal.com/braintree/help) referencing this issue.

We need a solid confirmation of this info so that we're correctly escalating the issue and that version 2.2.7-2 of Cardinal's SDK indeed does not resolve these alerts. Thanks all for your patience as we try to resolve this 🙏

I also want to mention that we are actively trying to reproduce these notifications with a demo app, too, but in the interest of the urgency of this issue we're asking for your help in confirming a few things. Thanks!

**Comment by hollabaq86** [comment]
for internal tracking, issue 1456

**Comment by bm185178** [comment]
We are gonna try it out and let you know if we ever get back from google.

**Comment by josephyanks** [comment]
Just a heads up, we had this issue as well and it applies to ANY version of the app that is on the play store that has the issue. In our case, we had a beta + dev application versions that we weren't actively updating that had the old sdk. We had to update those versions to clear the warning. Also, if your last released version to any of the tracks have this old version, you should either release into those tracks or purge the tracks of the violating releases as those will trigger the warnings as well. Google play support was extremely unhelpful, as most of their responses seem to be automated.

**Comment by 4gus71n** [comment]
Hi 👋 – The only Braintree lib our team is using is:

```
apply plugin: 'com.android.application'

// ...

repositories {
  maven { url "https://jitpack.io" }
  maven {
    url "https://cardinalcommerceprod.jfrog.io/artifactory/android"
    content {
      includeGroup "com.braintreepayments.api"
      includeGroup "org.jfrog.cardinalcommerce.gradle"
    }
    credentials {
      username 'braintree_team_sdk'
      // ...
    }
  }
}

dependencies {
  // ...
  implementation 'com.braintreepayments.api:drop-in:6.2.0'
  // ...
}
```

And we are still getting the warning from Google saying that we are not using Cardinal 2.2.7-2 – any idea if there's a new version of this API that we can use? Thanks! 🙇

**Comment by sshropshire** [comment]
Hi @4gus71n if you update your app to use the latest DropIn version [6.5.1](https://github.com/braintree/braintree-android-drop-in/blob/master/CHANGELOG.md#651) you should be good.

**Comment by matejhacin** [comment]
Hi,

I am not using Braintree DropIn dependency, so upgrading that dependency (as other answers here suggest) won't do anything for me. It seems like the problematic dependency in my case is `three-d-secure:3.11.1`.

<img width="369" alt="Screenshot 2022-12-19 at 18 08 29" src="https://user-images.githubusercontent.com/10889496/208481052-7921d62b-2196-455d-8e1b-e45afdb1f3ab.png">

I have not yet managed to do the migration to v4. It was on the roadmap for early 2023 but was hoping I could stay on v3 for a little while longer.

Do I have any other options here or is this gonna force me to do the migration to V4 earlier than planned?

Thanks

**Comment by sshropshire** [comment]
Hi @matejhacin we will add a `3.x` update soon after the holidays with a Cardinal update to give merchants more time to migrate. In the meantime, please consider upgrading to `v4` which is under active development.

**Comment by rhyskentish** [comment]
Hi @sshropshire just wondering if you could provide an update regarding the 3.x update for this? I can see a draft PR for it however do you have a timeline on when we could expect to see this? Thanks.

**Comment by sshropshire** [comment]
Hey @rhyskentish we've released version [3.20.0](https://github.com/braintree/braintree_android/blob/master/CHANGELOG.md#3200) with an up to date version of Cardinal.

**Comment by jamesyorke** [comment]
We are still hitting this issue even using Braintree SDK 4.39.0, 
Even though the Cardinal Mobile SDK version being used is now `org.jfrog.cardinalcommerce.gradle:cardinalmobilesdk:2.2.7-5` aka. above the minimum `2.2.7-2` the Google Play tells us it at least needs.

It seems to be the `com.braintreepayments.api:three-d-secure:4.39.0` that uses cardinal mobile sdk, and hence is the problematic one.
